pipeline_games/load_games.py

```diff
@@ -0,0 +1,289 @@
+            conn.rollback()
+            cur.close()
+
+
+def execute_batch_columns_for_genres(conn, data: pd.DataFrame, table: str, page_size=100) -> None:
+    """batch execution of adding genres into the database"""
+    tuples = [tuple(x) for x in data.to_numpy()]
+    cols = ','.join(list(data.columns))
+    query = "INSERT INTO %s(%s) VALUES(%%s,%%s) " % (table, cols)
+    with conn.cursor(cursor_factory=RealDictCursor) as cur:
+        try:
+            execute_batch(cur, query, tuples, page_size)
+            conn.commit()
+            cur.close()
+            print("execute_batch() done")
+        except (Exception, DatabaseError) as error:
+            print(f"Error: {error}")
+            conn.rollback()
+            cur.close()
+
+
+def execute_batch_columns_for_games(conn, data: pd.DataFrame, table: str, page_size=100) -> None:
+    """batch execution of adding games into the database"""
+    tuples = [tuple(x) for x in data.to_numpy()]
+    cols = ','.join(list(data.columns))
+
+    query = "INSERT INTO %s(%s) VALUES(%%s,%%s,%%s,%%s,%%s,%%s)" % (
+        table, cols)
+    with conn.cursor(cursor_factory=RealDictCursor) as cur:
+        try:
+            execute_batch(cur, query, tuples, page_size)
+            conn.commit()
+            cur.close()
+            print("execute_batch() done")
+        except (Exception, DatabaseError) as error:
+            print(f"Error: {error}")
+            conn.rollback()
+            cur.close()
+
+
+def get_existing_data(variable: str, table: str, value_name: str, value: str, conn: connect, cache: dict) -> None:
+    """Retrieves the existing data and adds to a cache dict given a provided value and table"""
+    if value in cache.keys():
+        return cache[value]
+    try:
+        with conn.cursor(cursor_factory=RealDictCursor) as cur:
+            cur.execute(
+                sql.SQL("""SELECT {variable} FROM {table} WHERE {value_name} = %s;""").format(
+                    variable=sql.Identifier(variable), table=sql.Identifier(table),
+                    value_name=sql.Identifier(value_name)), [value])
+            id_value = cur.fetchone()[variable]
+            cache[value] = id_value
+            cur.close()
+            return cache[value]
+    except TypeError:
+        return "None"
+
+
+def get_existing_data_for_genre(genre_c: str, user_c: bool, conn: connect, cache: dict) -> int | None:
+    """Retrieves the existing data for genre using cache and return genre_id"""
+    value = f'{genre_c} {user_c}'
+    if value in cache.keys():
+        return cache[value]
+    try:
+        with conn.cursor(cursor_factory=RealDictCursor) as cur:
+            cur.execute("""SELECT genre_id FROM genre WHERE genre = %s AND user_generated = %s;""", [
+                        genre_c, user_c])
+            id_value = cur.fetchone()['genre_id']
+            cache[value] = id_value
+            cur.close()
+            return cache[value]
+    except TypeError:
+        return "None"
```

Comment:
why this and more above are string None values?

Reply:
deleted so no longer relevant, but I would search the dataframe for where the id is none and insert these values into the database